lib/rules/space-infix-ops.js

```diff
@@ -21,49 +40,96 @@ module.exports = {
         schema: [
             {
                 type: "object",
-                properties: {
-                    int32Hint: {
-                        type: "boolean"
-                    }
-                },
+                properties: schemaProperties,
                 additionalProperties: false
             }
         ]
     },
 
     create(context) {
-        const int32Hint = context.options[0] ? context.options[0].int32Hint === true : false;
+        const option = context.options[0] || {};
+        const int32Hint = !!option.int32Hint;
+        const allSetting = option.all || "always";
 
-        const OPERATORS = [
-            "*", "/", "%", "+", "-", "<<", ">>", ">>>", "<", "<=", ">", ">=", "in",
-            "instanceof", "==", "!=", "===", "!==", "&", "^", "|", "&&", "||", "=",
-            "+=", "-=", "*=", "/=", "%=", "<<=", ">>=", ">>>=", "&=", "^=", "|=",
-            "?", ":", ",", "**"
-        ];
+        /*
+         * Build maps of operators that must or must not
+         * have spaces around them.
+         */
+        const spacesYes = { in: true, instanceof: true };
+        const spacesNo = {};
+
+        OPERATORS.forEach(operator => {
+            const thisSetting = option[operator];
+
+            if (thisSetting === "always" ||
+                (allSetting === "always" && !thisSetting)) {
+                spacesYes[operator] = true;
+            } else if (thisSetting === "never" ||
+              (allSetting === "never" && !thisSetting)) {
+                spacesNo[operator] = true;
+            }
+        });
 
         const sourceCode = context.getSourceCode();
 
         /**
+         * Determines whether a token appears to be an identifier.
+         * TODO: Is this the best way to determine this?
+         * TODO: This approach was copied from no-extra-parens.js line 283.
+         * @param {ASTNode} token - the token to be tested
+         * @returns {boolean} indicating whether it appears to be an identifier
+         */
+        function startsWithIdentifierPart(token) {
+            const text = sourceCode.getText(token);
+
+            return esUtils.code.isIdentifierPartES6(text.charCodeAt(0));
+        }
+
+        /**
          * Returns the first token which violates the rule
          * @param {ASTNode} left - The left node of the main node
          * @param {ASTNode} right - The right node of the main node
          * @returns {Object} The violator token or null
          * @private
          */
-        function getFirstNonSpacedToken(left, right) {
+        function getFirstViolatingToken(left, right) {
             const tokens = sourceCode.getTokensBetween(left, right, 1);
 
             for (let i = 1, l = tokens.length - 1; i < l; ++i) {
```

Comment:
I tried to make this change, but it caused an error in the tests. I think it's because this line assumes the first token is not the one be checked.
```js
                    const endOfPrev = tokens[i - 1].range[1];
```
This of course doesn't work when `i` is `0`.